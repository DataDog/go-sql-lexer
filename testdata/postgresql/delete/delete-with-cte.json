{
  "input": "WITH deleted AS (\n  DELETE FROM users WHERE last_login < NOW() - INTERVAL '2 years' RETURNING *\n)\nSELECT * FROM deleted;",
  "outputs": [
    {
      "expected": "WITH deleted AS ( DELETE FROM users WHERE last_login < NOW ( ) - INTERVAL ? RETURNING * ) SELECT * FROM deleted",
      "statement_metadata": {
        "size": 17,
        "tables": [
          "users"
        ],
        "commands": [
          "DELETE",
          "SELECT"
        ],
        "comments": [],
        "procedures": []
      }
    }
  ]
}