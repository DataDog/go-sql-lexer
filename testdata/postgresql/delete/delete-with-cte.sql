WITH deleted AS (
  DELETE FROM users WHERE last_login < NOW() - INTERVAL '2 years' RETURNING *
)
SELECT * FROM deleted;
