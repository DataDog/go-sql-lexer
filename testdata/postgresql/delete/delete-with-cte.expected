[
    {
        "expected": "WITH deleted AS ( DELETE FROM users WHERE last_login < NOW ( ) - INTERVAL ? RETURNING * ) SELECT * FROM deleted",
        "statement_metadata": {
            "size": 24,
            "tables": [
                "users",
                "deleted"
            ],
            "commands": [
                "DELETE",
                "SELECT"
            ],
            "comments": [],
            "procedures": []
        }
    }
]